libirods_rule_engine_plugin-indexing.cpp

```diff
@@ -44,18 +63,60 @@ int _delayExec(
 
 namespace {
 
+    // For handling of atomic metadata ops
+
+    using metadata_tuple = std::tuple<std::string, std::string, std::string>;
+    std::map <std::string, std::set<metadata_tuple>> atomic_metadata_tuples {};
+    std::string atomic_metadata_obj_path {};
+
+    // Other objects with visibility in this module only
+
+    std::set<std::string> indices_for_rm_coll;
     bool collection_metadata_is_new = false;
     std::unique_ptr<irods::indexing::configuration>     config;
     std::map<int, std::tuple<std::string, std::string>> opened_objects;
 
-    std::map < std::string,
-               std::vector < std::tuple < std::string, // id of   (sub-)coll
-                                          std::string, // path of (sub-)coll
-                                          std::string, /*A*/
-                                          std::string, /*V*/
-                                          std::string  /*U*/ >>> avus_to_purge;
+    const char* rm_force_kw = "*"; // default value tested for in "*_post" PEPs
+
+    auto get_indices_for_delete_by_query (rsComm_t& comm, const std::string &_object_name, bool recurs) -> std::set<std::string>
+    {
+        using irods::indexing::parse_indexer_string;
+        std::set<std::string> indices;
+        namespace fs   = irods::experimental::filesystem;
+        namespace fsvr = irods::experimental::filesystem::server;
+        using     fsp  = fs::path;
+        fsp node {_object_name}, up_node {_object_name};
+        auto coll_name {_object_name};
+        auto myQuery = [&]{
```

Comment:
make `coll_name` a parameter  of the myQuery lambda
also maybe myQuery -> getIndicesFromCollnAVUs